### ISSUE TITLE
Feature request: "localize" address format for USA

### ISSUE THREAD

**Issue by alyonapinchuk** [osmandapp/Osmand#6824]
The address format presented in the app is not suitable for searching for places in the US. Please add a corresponding address format to improve the user experience.

**Comment by gjedeer** [comment]
What's the correct format? How does it differ from the currently used format?

**Comment by Zian** [comment]
The national postal service ([United States Postal Service](https://pe.usps.com/text/pub28/28c2_001.htm#ep526236)) expects addresses to look like this:
ABC MOVERS
1500 E MAIN AVE STE 201
SPRINGFIELD VA 22162-1010

Most humans will write addresses using proper capitalization like this:
ABC Movers
1500 E Main Ave Ste 201
Springfield, VA 22162-1010

People may also write the address in a single line when searching in Google Maps or Bing Maps. A single-line entry would look like this:
ABC Movers, 1500 E Main Ave, Suite 201, Springfield, VA 22162-1010

Or, like this:
1500 E Main Ave, Suite 201, Springfield, VA 22162-1010

Finally, acronyms and abbreviations may be written out in full:
1500 East Main Avenue, Suite 201, Springfield, VA 22162-1010

And suite numbers may be omitted:
1500 E Main Ave, Springfield, VA 22162-1010

As well as ZIP codes (aka "postal codes"):
1500 East Main Avenue, Springfield, VA

I linked to the full USPS Postal Addressing Standards in the first paragraph of this post.

**Comment by zyphlar** [comment]
This is the #1 issue preventing me from easily using OsmAnd. I concluded that it was impossible to search by address the other day until someone told me the proper way to search is City, (click), Street, (click), Number. Which still only worked 80% of the time for me (online search was more reliable but still sometimes failed or did not appear.)

Localizing these formats either based on the jurisdiction being searched in or locale set in the app would improve adoption significantly. (Or, alternately, create more explanatory UX around the search flow instead of presuming what users will do/understand.)

**Comment by sonora** [comment]
I had suggested it somewhere already: a self explanatory search template screen may be the global answer, where you simply have fields for city, strret, zip code, etc. You fill out what you know and hit Search - simple as that. :)

**Comment by vshcherb** [comment]
@sonora Yes but we support 2 main formats:
1. Street, House number (City optional)
2. City, Street House number

I think the addresses in these issues are important for copy / paste from website or these are postal addreses which include "Company Name", Postal code, Apartment number i.e. lots extra information which doesn't help to find it in OpenStreetMap and only distracts.

So in that issue I think it is extremely important to "parse" query to understand where is City, Street and house number

**Comment by vshcherb** [comment]
Cause I don't live in US it is still hard to understand how to locate this house:
"1500 E Main Ave, Springfield, VA 22162-1010" hm... I even couldn't understand in which exactly city it is

**Comment by zyphlar** [comment]
vshcherb: if you can understand the format of Deutchland, Berlin, Alexanderplatz 12-345, then you can also understand 12-345 Alexanderplatz, Berlin, Deutchland. The difference is regional and customary, but to people accustomed to one format the other is a mystery. You and Sonora are both correct, parsing and pattern matching is necessary to know that "VA" is likely the US state of Virginia and not a road or city name. Likewise there is no city named 12-345 so it's pretty obvious that another parsing format is necessary.

The OSM website parses US-style addresses just fine, like Google Maps does, so it must be possible for us even if difficult.

**Comment by sonora** [comment]
I think Victor exactly demonstrates the point, not quite knowing which is the house number, the zip  ode, the apartment number etc. That's where the template would help, if the user knows. I did not mean excllusively, of course, but in addition to offering a google-like search slot.

**Comment by vshcherb** [comment]
Looking at concrete examples:
"345 Spear St, San Francisco, CA 94105, USA" - doesn't work
"345 Spear St, San Francisco" - works
So looks like address search gets quite confused with extra unnecessary information

**Comment by sonora** [comment]
Yes exactly, a single unknown abbreviation contained in a search string can cause no hits being displayed, that's one reason why there are so many complaints about it ever since we changed to the 'quick search'.

**Comment by sonora** [comment]
It's indeed a valid example, because we have most of the data:
- If you do an OsmAnd postcode search for _72212_, is shows you the correct suburb.
- Or if you perform a structured address search using button "_First specify city/town/locality_", and go to "_Little Rock_", then "_Sugar Maple Road_", it will take you to the correct street.
- Only the house number _4600_ is not in our data, as it's missing in OSM.

So why does searching for "_4600 Sugar Maple Lane, Little Rock, Arkansas, 72212_" fail so miserably? I think because it is full of very frequently occurring expressions like "Sugar", "Maple", "Lane", "Little", "Rock", and "Arkansas", all of which per se appear in countless potential search hits.

And I guess we may not group these using the commas supplied, i.e. take advantage of the fact that a hit for the combinations like "Sugar Maple Lane" or "Little Rock" are to be ranked much higher than occurrences of the single expressions like "Sugar" or "Arkansas", and the occurrence of both combinations in one hit still much higher? So maybe there is a way by rating exact hits of combinations of words as grouped by the user, even if we do not know the exact syntax or order of street/city/state/zip-code the user uses?

**Comment by zyphlar** [comment]
A good search engine should prioritize exact phrase matches over random order word matches, but aside from that it would be prudent to include a database of which regions have which address ordering scheme. A main reason I don't use OsmAnd regularly is because it expects addresses like this to be entered like Little Rock, Sugar Maple Lane, 4600. Which may work for some regions but is absolutely backwards in America. So a very simple fix would be to allow reverse order if the viewport or GPS location or system locale setting is American. (Other countries surely have their own ordering as well, so this can benefit everyone.)

Organic Maps handles these searches much better due to tokenizing and match scoring, however it will fail similarly by only showing you the searched street since 4600 isn't in OSM. You can work around this by downloading OpenSuperMaps for OsmAnd which includes Openaddress data within it.

So there's two issues with this Arkansas case: OsmAnd's search ordering, and also a lack of data in OSM which means we either need additional OsmAnd data sources or further OSM import work for local addresses.

**Comment by davew99** [comment]
Thank you Zyphlar for the referral over to OpenSuperMaps? And, thank you to the gentleman running that site.  I downloaded the address files from OpenSuperMaps and OsmAnd was actually able to find some addresses in USA format that OsmAnd was not able to find previously.  Only time will tell how much better these address files make the overall OsmAnd address search overall, but I am hopeful.
Now the question is, can OsmAnd just incorporate these address files directly into OsmAnd so that we do not need to download and install these address files all the time?  I am sure that the gentleman who is running that site is eventually going to get tired of maintaining the files and site.

**Comment by zyphlar** [comment]
OpenSuperMap is a merging of OSM data and OpenAddresses.io data so fortunately it's not all a single person project but I do agree that all these projects are worth maintaining and supporting and that OsmAnd would benefit from incorporating it into their build chain by default (or at least offering it as an alternative download/overlay)

**Comment by vshcherb** [comment]
Yes consolidating with OpenSuperMap is a right approach. As we don't have enough man power yet and focused on routing questions, search lacks a bit behind. Hopefully will switch soon

**Comment by ENLinden** [comment]
What would be the first steps to contribute towards closing this issue? I think the entire search setup needs overhauled and simplified from a UX POV to increase adoption, but I think this is a good place to start and am eager to contribute if I could be pointed in the right direction.

**Comment by vshcherb** [comment]
I think we've recently fixed main issue with search by adding states / abbrevation ot region maps. 
So finally `"4600 Sugar Maple Lane, Little Rock, Arkansas, 72212"`. 

As issue is very old the conversation may be went away from original request. However we will close and move to testing if you have any open issues, please follow up.

**Comment by davew99** [comment]
Hi,Sorry, but appears to me that nothing has been fixed. I still get a large percentage (way over half) of searches in standard US format that OSM cannot find. Such as:35th Street,  Manhatten Beach, Ca 90266.When I try this address I get nothing but street names that have 'Manhattan' in them, but even at that they are not in California. Put the same address in Google and it will find the correct street.It still not clear to me why OSM can not find a simple US street address? It is so powerful otherwise? 
-------- Original message --------From: vshcherb ***@***.***> Date: 10/11/25  5:56 AM  (GMT-08:00) To: osmandapp/OsmAnd ***@***.***> Cc: davew99 ***@***.***>, Comment ***@***.***> Subject: Re: [osmandapp/OsmAnd] Feature request: "localize" address format for
  USA (#6824) vshcherb left a comment (osmandapp/OsmAnd#6824)
I think we've recently fixed main issue with search by adding states / abbrevation ot region maps.
So finally "4600 Sugar Maple Lane, Little Rock, Arkansas, 72212".
As issue is very old the conversation may be went away from original request. However we will close and move to testing if you have any open issues, please follow up.

**Comment by DmitryAlexei** [comment]
@davew99 have you tried search in Nightly version or latest beta? (https://osmand.net/docs/versions/nightly_versions#nightly-builds)

**Comment by davew99** [comment]
I only use what has been released via the play store. 
-------- Original message --------From: DmitryAlexei ***@***.***> Date: 10/20/25  5:31 AM  (GMT-08:00) To: osmandapp/OsmAnd ***@***.***> Cc: davew99 ***@***.***>, Mention ***@***.***> Subject: Re: [osmandapp/OsmAnd] Feature request: "localize" address format for
  USA (#6824) DmitryAlexei left a comment (osmandapp/OsmAnd#6824)
@davew99 have you tried search in Nightly version or latest beta? (https://osmand.net/docs/versions/nightly_versions#nightly-builds)

**Comment by davew99** [comment]
I have signed up for the beta, I will let you know. 
-------- Original message --------From: DmitryAlexei ***@***.***> Date: 10/20/25  5:31 AM  (GMT-08:00) To: osmandapp/OsmAnd ***@***.***> Cc: davew99 ***@***.***>, Mention ***@***.***> Subject: Re: [osmandapp/OsmAnd] Feature request: "localize" address format for
  USA (#6824) DmitryAlexei left a comment (osmandapp/OsmAnd#6824)
@davew99 have you tried search in Nightly version or latest beta? (https://osmand.net/docs/versions/nightly_versions#nightly-builds)

**Comment by Zirochkabila** [comment]
# Testing 1

> OsmAnd~ 5.3.0#6347m, Released: 2025-10-27
> Web-prod
> 5.2.2.5265


Based on the search results, the location is either not found (OSM, Nominatim) or is displayed as found elsewhere (OsmAnd - results on all three platforms differ).

<details><summary>Details</summary>
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/4b62c91b-afc3-4857-b343-30a7d5560720"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/c31e41bb-56fb-463c-92bc-e11bcf87a85e"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/1d4efc00-84e0-4d8e-a8b5-31c9c6d4df03"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/3177e7c4-ad51-42b6-ad7c-6839a5a2af0a"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/fbfdbf57-5659-494a-8a0b-36223b347978"  width="300" height="auto"/> </th>
  </tr>
</table>  
</details>
